core/src/test/java/org/apache/calcite/test/SqlValidatorDynamicTest.kt

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.test
+
+import org.apache.calcite.rel.type.RelDataTypeFactory
+import org.apache.calcite.sql.test.SqlTestFactory
+import org.apache.calcite.sql.test.SqlTester
+import org.apache.calcite.sql.test.SqlValidatorTester
+import org.apache.calcite.test.catalog.MockCatalogReaderDynamic
+import org.apache.calcite.testlib.annotations.LocaleEnUs
+import org.junit.jupiter.api.Test
+
+/**
+ * Concrete child class of [SqlValidatorTestCase], containing lots of unit
+ * tests.
+ *
+ * If you want to run these same tests in a different environment, create a
+ * derived class whose [getTester] returns a different implementation of
+ * [SqlTester].
+ */
+@LocaleEnUs
+class SqlValidatorDynamicTest : SqlValidatorTestCase() {
+    /**
+     * Dynamic schema should not be reused since it is mutable, so
+     * we create new SqlTestFactory for each test
+     */
+    override fun getTester(): SqlTester = SqlValidatorTester(SqlTestFactory.INSTANCE
+        .withCatalogReader { typeFactory: RelDataTypeFactory, caseSensitive: Boolean ->
+            MockCatalogReaderDynamic(
+                typeFactory,
+                caseSensitive
+            )
+        }
+    )
+
+    /**
+     * Test case for
+     * [Dynamic Table / Dynamic Star support](https://issues.apache.org/jira/browse/CALCITE-1150).
+     */
+    @Test
```

Comment:
There is something wrong with this test format. It should be that @Test and the function name are on the same line.E.g: @Test fun `ambiguous dynamic star`()

Reply:
Technically speaking, there's Kotlin official style guide, and formatting `@Test` into the same line as `fun` would violate the official guide.

In other words, if I format `@Test` as  `@Test fun ...`, then IDE would split them back into individual lines, so I suggest we should stick to the official formatting style.

Reply:
Just in case, `@Override` in Kotlin is `override fun` (see above) which is very close to Calcite's good and old `@Override must be in the same line` rule.

Reply:
> Just in case, `@Override` in Kotlin is `override fun` (see above) which is very close to Calcite's good and old `@Override must be in the same line` rule.

OK, I see, thank you very much for your explanation. I see that in addition to the core module, there are other modules that are still tests of junit4. Are there plans to upgrade to junit5?

Reply:
junit5 is enabled for all the modules, so junit5 sets can be used everywhere.

However, there are still non-trivial tests written in JUnit4 style (e.g. elastic module), and it might take certain effort to migrate them.

> I see that in addition to the core module

I think all the tests in `core` are JUnit5 based.

Most of the tests have been migrated to junit5, and the only modules that have some of the tests in junit4 are `cassandra, elasticsearch, geode, linq4j, and mongodb`:  https://github.com/apache/calcite/search?l=Java+Properties&q=junit4 
Feel free to migrate the existing

Reply:
> junit5 is enabled for all the modules, so junit5 sets can be used everywhere.
> 
> However, there are still non-trivial tests written in JUnit4 style (e.g. elastic module), and it might take certain effort to migrate them.
> 
> > I see that in addition to the core module
> 
> I think all the tests in `core` are JUnit5 based.
> 
> Most of the tests have been migrated to junit5, and the only modules that have some of the tests in junit4 are `cassandra, elasticsearch, geode, linq4j, and mongodb`: https://github.com/apache/calcite/search?l=Java+Properties&q=junit4
> Feel free to migrate the existing

Well, I will migrate the tests according to the module and upgrade from junit4 to junit5.